### PR TITLE
Add description to courses

### PR DESCRIPTION
**Pull request by tjingsheng**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -179,6 +181,21 @@ public static LocalUnit parseLocalUnit(String localUnit) throws ParseException {
...
+    /**
+     * Parses a {@code String localDescription} into an {@code LocalDescription}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code localDescription} is invalid.
+     */
+    public static LocalDescription parseLocalDescription(String localDescription) throws ParseException {
+        requireNonNull(localDescription);
+        String trimmedLocalDescription = localDescription.trim();
+        if (!PartnerDescription.isValidPartnerDescription(trimmedLocalDescription)) {
```
Comment (dlathyun) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:193]: Local**

File: src/main/java/seedu/address/model/localcourse/LocalCourse.java [hunk 2]
```diff
@@ -79,22 +87,22 @@ public boolean equals(Object other) {
 
         LocalCourse otherLocalCourse = (LocalCourse) other;
         return localCode.equals(otherLocalCourse.localCode)
-                && localName.equals(otherLocalCourse.localName)
-                && localUnit.equals(otherLocalCourse.localUnit);
+            && localName.equals(otherLocalCourse.localName)
+            && localUnit.equals(otherLocalCourse.localUnit);
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/localcourse/LocalCourse.java:91]: Do we need to compare localDescription as well?

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 3]
```diff
@@ -120,22 +147,25 @@ public static ReadOnlyUniversityCatalogue getSampleUniversityCatalogue() {
...
+                new PartnerCode("INFR1101"),
+                new PartnerName("Computer Security"),
+                new PartnerUnit(5.0),
+                new PartnerDescription("Fun security module.")),
+            new PartnerCourse(
+                new University(new UniversityName("Harvard University")),
+                new PartnerCode("COM3041"),
+                new PartnerName("Computer Networks"),
+                new PartnerUnit(2.0),
+                new PartnerDescription("Fun network module."))};
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:168]: Will have merge conflict with cheng hours latest PR :(

File: src/main/java/seedu/address/seplendidui/PartnerCourseItem.java [hunk 4]
```diff
@@ -33,5 +35,6 @@ public PartnerCourseItem(PartnerCourse partnerCourse, int displayedIndex) {
         id.setText(displayedIndex + ".");
         partnerName.setText(partnerCourse.getPartnerName().toString());
         partnerCode.setText(partnerCourse.getPartnerCode().toString());
+        partnerUniversity.setText(partnerCourse.getPartnerUniversity().getUniversityName().toString());
```
Comment (dlathyun) [inline src/main/java/seedu/address/seplendidui/PartnerCourseItem.java:38]: I think you can straightaway call toString() on university

File: src/test/data/JsonLocalCourseCatalogueStorageTest/invalidLocalCourseCatalogue.json [hunk 5]
```diff
@@ -2,7 +2,8 @@
   "localCourses": [
     {
       "localCode": "INVALID1234567890",
-      "localName": " "
+      "localName": " ",
+      "localUnit" : "a"
```
Comment (dlathyun) [inline src/test/data/JsonLocalCourseCatalogueStorageTest/invalidLocalCourseCatalogue.json:6]: Do we need localDescription here?

**Review by dlathyun — CHANGES_REQUESTED** [review]
LGTM except few minor typos

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 6]
```diff
@@ -17,9 +17,11 @@ public class CliSyntax {
     public static final SeplendidParameter PARAMETER_LOCALCODE = new SeplendidParameter("localcode");
     public static final SeplendidParameter PARAMETER_LOCALNAME = new SeplendidParameter("localname");
     public static final SeplendidParameter PARAMETER_LOCALUNIT = new SeplendidParameter("localunit");
+    public static final SeplendidParameter PARAMETER_LOCALDESCRIPTION = new SeplendidParameter("localdescription");
     public static final SeplendidParameter PARAMETER_PARTNERCODE = new SeplendidParameter("partnercode");
     public static final SeplendidParameter PARAMETER_PARTNERNAME = new SeplendidParameter("partnername");
     public static final SeplendidParameter PARAMETER_PARTNERUNIT = new SeplendidParameter("partnerunit");
+    public static final SeplendidParameter PARAMETER_PARTNERDESCRIPTION = new SeplendidParameter("partnerdescription");
```
[the remark this answers is not in the document]
Comment (tjingsheng) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:24]: no

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 3 above)
Comment (tjingsheng) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:168]: :P

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (tjingsheng) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:193]: good catch

File: src/main/java/seedu/address/model/localcourse/LocalCourse.java (hunk 2 above)
Comment (tjingsheng) [reply src/main/java/seedu/address/model/localcourse/LocalCourse.java:91]: yes

File: src/main/java/seedu/address/seplendidui/PartnerCourseItem.java (hunk 4 above)
Comment (tjingsheng) [reply src/main/java/seedu/address/seplendidui/PartnerCourseItem.java:38]: i think cannot

**Review by dlathyun — APPROVED** [review]
LGTM

**Pull request merged**